### ISSUE TITLE
Error using prettier-plugin-solidity with Vite and Monaco

### ISSUE THREAD

**Issue by folego** [prettier-solidity/prettier-plugin-solidity#942]
I am building a Solidity editor using Monaco in a ReactJS App with Vite. I am facing some issues trying to format the code.

**First approach:**
I tried to import the prettier-solidity-plugin installing via NPM and using:
```
import prettier from 'prettier/standalone';
import solidityPlugin from 'prettier-plugin-solidity';
```

Got this error in the browser and during the build:
`"createRequire" is not exported by "__vite-browser-external", imported by "node_modules/prettier/internal/internal.mjs".
file: C:/Code/Shepherd/ui/node_modules/prettier/internal/internal.mjs:1:9
1: import { createRequire as __prettierCreateRequire } from "module";
            ^
2: import { fileURLToPath as __prettierFileUrlToPath } from "url";
3: import { dirname as __prettierDirname } from "path";
error during build:
RollupError: "createRequire" is not exported by "__vite-browser-external", imported by "node_modules/prettier/internal/internal.mjs".`

**Second approach:**
Using `<script>` in the index.html file as described in the docs.

Got this error:
`ConfigError: Couldn't resolve parser "solidity-parse". Plugins must be explicitly added to the standalone bundle.`

Code Sample using `<script>` (for import is almost the same, just replace the `prettierPlugins `for `solidityPlugin`:
```
const formattedCode = await prettier.format(sourceCode, {
	parser: "solidity-parse",
	plugins: [prettierPlugins],
	singleQuote: false,
	printWidth: 200,
	tabWidth: 4,
	useTabs: false,
	bracketSpacing: false,
});
```

Is there anything I am missing? Thanks in advance.

**Comment by Janther** [comment]
For the browser solution, if the file has been properly loaded, `prettierPlugins` should be populated with the solidity plugin already.
You could log the values of `solidityPlugin` and `prettierPlugins`, this might help find the issue.

For the NPM approach, I see you are using `prettier/standalone`.
You could try to load also our standalone file:

```Javascript
import prettier from 'prettier/standalone';
import solidityPlugin from 'prettier-plugin-solidity/dist/standalone.cjs';
```

Another approach is to tell `vite` to always load the browser friendly files.
I haven't used `vite` but having a quick look at their documentation, they support `resolve.alias` in their configuration.

Could you see if this works:
```Javascript
resolve: {
  alias: [
    { find: 'prettier', replacement: 'prettier/standalone' },
    { find: 'prettier-plugin-solidity/', replacement: 'prettier-plugin-solidity/dist/standalone.cjs' }
  ]
}
```

Then you should be able to just import 'prettier' and 'prettier-plugin-solidity' within your project.

**Comment by folego** [comment]
Thanks again for your time.

I finally got this working using the standalone.js (not the cjs). You should add to docs to explain this, I think it will help other people.

Follow my solution:

vite.config.js
```
resolve: {
            alias: {
                "prettier": "prettier/standalone",
                "prettier-plugin-solidity/": "prettier-plugin-solidity/dist/standalone.js",
            },
        },
```

My App File:
```
import prettier from 'prettier';
import solidityPlugin from 'prettier-plugin-solidity';

                    const formattedCode = await prettier.format(model.getValue(), {
                        parser: "solidity-parse",
                        plugins: [solidityPlugin],
                        singleQuote: false,
                        printWidth: 200,
                        tabWidth: 4,
                        useTabs: false,
                        bracketSpacing: false,
                    });
```

Also the build is working.

**Comment by Janther** [comment]
Oh yeah! 🤦 the next version will ship with a `.cjs` extension.

Glad you got it sorted and thank you for the suggestion of adding this particular case to the readme.

**Comment by Janther** [comment]
Hey @folego ,

We are trying a new build that doesn't need to be aliased. Could you help us test it?
just install `prettier-plugin-solidity@1.1.4-dev` and then just

```Javascript
import prettier from "prettier/standalone";
import pluginSolidity from "prettier-plugin-solidity/standalone";
```

**Comment by folego** [comment]
@Janther 

I replaced the old version with `npm install prettier-plugin-solidity@1.1.4-dev,` removed the alias in vite.config, and imported directly as described.
Everything working as expected🎉

**Comment by Janther** [comment]
@folego 
We just released a new official [version 1.2.0](https://github.com/prettier-solidity/prettier-plugin-solidity/releases/tag/v1.2.0) thank you for your feedback.
